### ISSUE TITLE
Too big bundle size (instead you can do this)

### ISSUE THREAD

**Issue by zardoy** [wyatt-herkamp/vue3-simple-icons#3]
Hey! I don't really understand what's the point of using Vite in library mode. Sometimes when you use transpilers in libraries instead of bundlers (shipping a single file) you provide much better dx and make it easier to do manual app bundle size optimizations.

I'd recommend using original `simple-icons` package with a classic way of embedding svgs in any framework, eg in Vue:

```vue
<script setup lang="ts">
import TwitterIcon from 'simple-icons/icons/x.svg?raw'  // import Twitter icon in Vite app
</script>
<template>
    <svg v-html="TwitterIcon" />
</template>
```

btw it would be great to have build instructions & working website here. I believe the original issue can be easily fixed by adding __NO_SIDE_EFFECTS__ or pure comments iirc

**Comment by wyatt-herkamp** [comment]
> Hey! I don't really understand what's the point of using Vite in library mode.
Same! When I do frontend development I just add random stuff until it "works". 

I don't understand why the bundle size is so big. As I would think any unused components should not be added to the final bundle. 

If that code works then you don't really need this library

**Comment by wyatt-herkamp** [comment]
This was a fork of another library that I just updated. So the design is not really mine.

**Comment by wyatt-herkamp** [comment]
May I ask what exactly you are proposing I change about the library?   
  
I am a Rust and Java dev. So my understanding of frontend is limited.

**Comment by zardoy** [comment]
Oh that's totally fine then. I think at least using these magic comments in the bundle might help, I'll look into it. Also I've got an idea of using simple-icons as peer dependency (user manages version) and generating vue components at install time so we don't ever need to update the package with new releases.

**Comment by jaythomas** [comment]
If a consuming app is tree shaking like it should then bundle size isn't an issue. However, distributing individual icon files that import from a base icon would be a nice-to-have option.

> I'd recommend using original `simple-icons` package with a classic way of embedding svgs in any framework, eg in Vue:
> 
> ```
> <script setup lang="ts">
> import TwitterIcon from 'simple-icons/icons/x.svg?raw'  // import Twitter icon in Vite app
> </script>
> <template>
>     <svg v-html="TwitterIcon" />
> </template>
> ```
> 

I don't recommend using v-html to trust code that you don't own.

**Comment by zardoy** [comment]
> If a consuming app is tree shaking like it should then bundle size isn't an issue. 

Can you rephrase?

From what I tested minimal app size with Vite is 500kb. I don't think this optimisation is nice to have option, bundle size was always a priority for web applications since it can affect load times.
> I don't recommend using v-html to trust code that you don't own.

The v-html tip is not relevant in the current context. If you look into build script you will see that it also inlines svg content. So if you think that simple-icons can contain malicious code, this package will be affected as well and won't matter how you use the bad icon: via this package or via v-html.

**Comment by wyatt-herkamp** [comment]
I could figure out how to do something similar to Font Awesome. 

Where you need to enable the icons you want. So it prevents them from all being bundled?

**Comment by jaythomas** [comment]
> > If a consuming app is tree shaking like it should then bundle size isn't an issue.
> 
> Can you rephrase?
> 
> From what I tested minimal app size with Vite is 500kb. I don't think this optimisation is nice to have option, bundle size was always a priority for web applications since it can affect load times.

What I mean is even if the bundle was 100MB tree shaking will give the consuming app a bundle that contains only the code for the icons you import. Inspecting `dist/vue3-simple-icons.mjs` I see each icon component is getting named exports:

```js
export {
  XT3 as AbbottIcon,
  QT3 as AbbrobotstudioIcon,
  YT3 as AbbvieIcon,
  ...
  gL4 as ZoiperIcon,
  fL4 as ZomatoIcon,
  pL4 as ZoomIcon,
  wL4 as ZorinIcon,
  IL4 as ZoteroIcon,
  ML4 as ZshIcon,
  HL4 as ZulipIcon,
  yL4 as ZyteIcon
};
```

so unless I missed something there is nothing troubling with the bundle size. The part I said is a nice-to-have would be having separate files for each icon component for apps that aren't configured to treeshake and would benefit from having the code broken out into separate physical files.

> > I don't recommend using v-html to trust code that you don't own.
> 
> The v-html tip is not relevant in the current context. If you look into build script you will see that it also inlines svg content. So if you think that simple-icons can contain malicious code, this package will be affected as well and won't matter how you use the bad icon: via this package or via v-html.

Can you explain to me where v-html is being used in this library? The icons in the build are statically defined as such:

```js
const ET3 = /* @__PURE__ */ z(d, [["render", v]]), g = o({
  name: "DotnetIcon",
  props: {
    size: {
      type: String,
      default: "24"
    }
  },
  setup(e) {
    return e.size === void 0 ? { size: "24" } : { size: e.size.slice(-1) === "x" ? e.size.slice(0, e.size.length - 1) + "em" : parseInt(e.size) + "px" };
  }
}), f = ["width", "height"], p = /* @__PURE__ */ s("title", null, ".NET", -1), w = /* @__PURE__ */ s("path", { d: "M24 8.77h-2.468v7.565h-1.425V8.77h-2.462V7.53H24zm-6.852 7.565h-4.821V7.53h4.63v1.24h-3.205v2.494h2.953v1.234h-2.953v2.604h3.396zm-6.708 0H8.882L4.78 9.863a2.896 2.896 0 0 1-.258-.51h-.036c.032.189.048.592.048 1.21v5.772H3.157V7.53h1.659l3.965 6.32c.167.261.275.442.323.54h.024c-.04-.233-.06-.629-.06-1.185V7.529h1.372zm-8.703-.693a.868.829 0 0 1-.869.829.868.829 0 0 1-.868-.83.868.829 0 0 1 .868-.828.868.829 0 0 1 .869.829Z" }, null, -1), I = [
  p,
  w
];
```

**Comment by wyatt-herkamp** [comment]
I am closing this issue. I recommend to anyone really worried about bundle size just use simple-icons directly. Either by serving the svg files as static files or something like that. 

If bundle size does become a real problem. We can look into adopting something similar to font awesome.

**Comment by zardoy** [comment]
tried this and seems there is no such problem (just a 7kb bundle):

```ts
import { AbbottIcon } from 'vue3-simple-icons'

console.log(AbbottIcon)
```
